src/lib/snark_pool/snark_pool.ml

```diff
@@ -59,24 +78,72 @@ end) :
     let fee (t : t) = t.fee
   end
 
-  type t = Priced_proof.t Work.Table.t [@@deriving sexp, bin_io]
+  type t =
+    { snark_table: Priced_proof.t Work.Table.t
+    ; mutable ref_table: int Work.Table.t option }
+  [@@deriving sexp, bin_io]
+
+  let removed_breadcrumb_wait = 10
```

Comment:
TODO: tweak this value and potentially give it a better name.. It's the amount of breadcrumbs-worth of garbage work the system allows the snark pool to accumulate before collecting it.